docs/reference/modules/discovery.asciidoc

```diff
@@ -1,30 +1,805 @@
+`master-a`, `master-b` and `master-c`) the configuration will look as follows:
+
+[source,yaml]
+--------------------------------------------------
+cluster.initial_master_nodes:
+  - master-a
+  - master-b
+  - master-c
+--------------------------------------------------
+
+Alternatively the IP addresses or hostnames
+(<<node.name,if node name defaults to the host name>>) can be used. If there
+is more than one Elasticsearch node with the same IP address or hostname then
+the transport ports must also be given
+
+[source,yaml]
+--------------------------------------------------
+cluster.initial_master_nodes:
+  - 10.0.10.101
+  - 10.0.10.102:9300
+  - 10.0.10.102:9301
+  - master-node-hostname
+--------------------------------------------------
+
+Like all node settings, it is also possible to specify the initial set of
+master nodes on the command-line that is used to start Elasticsearch:
+
+[source,bash]
+--------------------------------------------------
+$ bin/elasticsearch -Ecluster.initial_master_nodes=master-a,master-b,master-c
+--------------------------------------------------
+
+It is technically sufficient to set this on a single master-eligible node in
+the cluster, and only to mention that single node in the setting, but this
+provides no fault tolerance before the cluster has fully formed. It
+is therefore better to bootstrap using at least three master-eligible nodes.
+In any case, when specifying the list of initial master nodes, **it is vitally
+important** to configure each node with exactly the same list of nodes, to
+prevent two independent clusters from forming. Typically you will set this on
+the nodes that are mentioned in the list of initial master nodes.
+
+NOTE: In alpha releases, all listed master-eligible nodes are required to be
+  discovered before bootstrapping can take place. This requirement will be
+  relaxed in production-ready releases.
+
+WARNING: You must put exactly the same set of initial master nodes in each
+  configuration file (or leave the configuration empty) in order to be sure
+  that only a single cluster forms during bootstrapping and therefore to
+  avoid the risk of data loss.
+
+[float]
+==== Choosing a cluster name
+
+The `cluster.name` allows you to create multiple clusters which are separated
+from each other. Nodes verify that they agree on their cluster name when they
+first connect to each other, and if two nodes have different cluster names then
+they will not communicate meaningfully and will not belong to the same cluster.
+The default value for the cluster name is `elasticsearch`, but it is
+recommended to change this to reflect the logical name of the cluster.
+
+[float]
+==== Auto-bootstrapping in development mode
+
+If the cluster is running with a completely default configuration then it will
+automatically bootstrap based on the nodes that could be discovered within a
+short time after startup. Since nodes may not always reliably discover each
```

Comment:
It would be great to work some wordings that indicate that nodes will only discover node on the same machine (and thus it relatively safe and easy to get going )

Reply:
Good point, see 17be8bb2494cbbee7c52cdb5f120d2348b45a640.